src/Npgsql/TypeHandlers/CompositeHandlers/CompositeHandler.cs

```diff
@@ -90,25 +97,89 @@ public override int ValidateAndGetLength(T value, ref NpgsqlLengthCache? lengthC
             lengthCache.Set(0);
 
             // number of fields + (type oid + field length) * member count
-            var length = sizeof(int) + sizeof(int) * 2 * _memberHandlers.Length;
-            foreach (var member in _memberHandlers)
+            var memberHandlers = _memberHandlers!;
+            var length = sizeof(int) + sizeof(int) * 2 * memberHandlers.Length;
+            foreach (var member in memberHandlers)
                 length += member.ValidateAndGetLength(value, ref lengthCache);
 
             return lengthCache.Lengths[position] = length;
         }
 
-        CompositeMemberHandler<T>[] CreateMemberHandlers() => CreateMemberHandlers((PostgresCompositeType)PostgresType, _typeMapper, _nameTranslator);
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        void Initialize()
+        {
+            if (_memberHandlers is null)
+                InitializeCore();
+        }
 
-        static CompositeMemberHandler<T>[] CreateMemberHandlers(PostgresCompositeType pgType, ConnectorTypeMapper typeMapper, INpgsqlNameTranslator nameTranslator)
+        void InitializeCore()
         {
-            var clrType = typeof(T);
+            var pgType = (PostgresCompositeType)PostgresType;
             var pgFields = pgType.Fields;
-            var clrHandlers = new CompositeMemberHandler<T>[pgFields.Count];
-            var clrHandlerCount = 0;
 
-            var clrHandlerType = IsValueType<T>.Value
+            var clrType = typeof(T);
+            var clrMemberHandlers = new CompositeMemberHandler<T>[pgFields.Count];
+            var clrMemberHandlerCount = 0;
+            var clrMemberHandlerType = IsValueType<T>.Value
                 ? typeof(CompositeStructMemberHandler<,>)
                 : typeof(CompositeClassMemberHandler<,>);
+            var clrConstructorHandlerType = typeof(CompositeParameterHandler<>);
```

Comment:
This variable doesn't seem to be valuable, why not inline it

Reply:
Just a little optimization to stop type loading in each iteration, but mostly for similarity with the same method working with parameters.

Reply:
Are you sure it's actually an optimization? But OK, no problem.

Reply:
It could, but this mostly for similarity with member handler creation. They behave the same way and looks near the same.